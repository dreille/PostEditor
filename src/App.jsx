import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPost } from './Redux/config/title';
import Search from './Search';
import EditPost from './EditPost';

import './styles.css';

function App() {
  const dispatch = useDispatch();

  function getData() {
    dispatch(getPost());
  }
  useEffect(() => {
    getData();
  }, []);

  const user = useSelector((state) => state.posts.posts);

  return (
    <div className="App">
      <h1> Posts</h1>
      <div className="container">
        <div className="searchRes">
          {user.length !== 0 ? <Search data={user} /> : ''}
        </div>
        <div className="editZone">
          <EditPost />
        </div>
      </div>
    </div>
  );
}

export default App;
