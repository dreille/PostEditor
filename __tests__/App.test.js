import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../src/App';
import store from '../src/Redux/configureStore';

describe('This suit is to test the Body component', () => {
  test('Snapshot of App', () => {
    const { asFragment } = render(<Provider store={store}><App /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
