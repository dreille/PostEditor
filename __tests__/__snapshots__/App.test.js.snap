// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component Snapshot of App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
       Posts
    </h1>
    <div
      class="container"
    >
      <div
        class="searchRes"
      />
      <div
        class="editZone"
      >
        <div
          class="editContainer"
        >
          <div>
            <h1>
               Edit Post
            </h1>
          </div>
          <div>
            <form>
              <div>
                <div
                  class="postTitle"
                >
                  <label
                    class="title"
                  >
                    <h2>
                      Title:
                    </h2>
                  </label>
                  <input
                    class="title-box"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div>
                <div
                  class="postBody"
                >
                  <label
                    class="Body"
                  >
                    <h2>
                      Body:
                    </h2>
                  </label>
                  <textarea
                    class="body-box"
                  />
                </div>
              </div>
              <div
                class="edit-button-container"
              >
                <button
                  class="edit-button"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
